SU2_CFD/include/fluid/CIncIdealGasPolynomial.hpp

```diff
@@ -72,18 +77,68 @@ class CIncIdealGasPolynomial final : public CFluidModel {
     Temperature = t;
     Density = Pressure / (Temperature * Gas_Constant);
 
-    /* Evaluate the new Cp from the coefficients and temperature. */
+    /* Evaluate the new Cp and enthalpy from the coefficients and temperature. */
     Cp = coeffs_[0];
+    Enthalpy = coeffs_[0] * t - Enthalpy_Ref;
     su2double t_i = 1.0;
     for (int i = 1; i < N; ++i) {
       t_i *= t;
       Cp += coeffs_[i] * t_i;
+      Enthalpy += coeffs_[i] * t_i * t / (i + 1);
+    }
+    Cv = Cp / Gamma;
+  }
+
+  /*!
+   * \brief Set the Dimensionless State using enthalpy.
+   * \param[in] val_enthalpy - Enthalpy value at the point.
+   */
+  void SetTDState_h(su2double val_enthalpy, const su2double* val_scalars = nullptr) override {
+    Enthalpy = val_enthalpy;
+    /*--- convergence criterion for temperature in [K], high accuracy needed for restarts. ---*/
+    const su2double toll = 1e-5;
+    su2double temp_iter = 300.0;
+    su2double Cp_iter = 0.0;
+    su2double delta_temp_iter = 1e10;
+    su2double delta_enthalpy_iter;
+    const int counter_limit = 20;
+
+    int counter = 0;
+
+    /*--- Computing temperature given enthalpy using Newton-Raphson. ---*/
+    while ((abs(delta_temp_iter) > toll) && (counter++ < counter_limit)) {
+      /* Evaluate the new Cp and enthalpy from the coefficients and temperature. */
+      Cp_iter = coeffs_[0];
+      su2double Enthalpy_iter = coeffs_[0] * temp_iter - Enthalpy_Ref;
+      su2double t_i = 1.0;
+      for (int i = 1; i < N; ++i) {
+        t_i *= temp_iter;
+        Cp_iter += coeffs_[i] * t_i;
+        Enthalpy_iter += coeffs_[i] * t_i * temp_iter / (i + 1);
+      }
+
+      delta_enthalpy_iter = Enthalpy - Enthalpy_iter;
+
+      delta_temp_iter = delta_enthalpy_iter / Cp_iter;
+
+      temp_iter += delta_temp_iter;
+      if (temp_iter < 0.0) {
+        cout << "Warning: Negative temperature has been found during Newton-Raphson" << endl;
```

Comment:
```suggestion
      if (temp_iter < 0.0) {
        cout << "Warning: Negative temperature has been found during Newton-Raphson" << endl;
        temp_iter = 0.0;
```

Reply:
Hi!!

thank you so much for the comment, 
with the user option of using temperature limits for the incompressible solver, 
Would it be more consistent with that option to return the minimum temperature limit? 
the if statement would be: 
if (temp_iter < temp_limit_minimum){
     temp_iter = temp_limit_minimum
}
the default minimum temperature limit is 0.0

please let me know what you think about this.

Thank you so much in advance!!!

Reply:
> Hi!!
> 
> thank you so much for the comment, with the user option of using temperature limits for the incompressible solver, 
> Would it be more consistent with that option to return the minimum temperature limit? 

Yes, if there is a minimum temperature limit that you can use it would be better, by the time you reach T=0, you are probably already too far gone.